independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/ServerRequestFilter.java

```diff
@@ -47,6 +47,24 @@
  * <li>{@link ResourceInfo}
  * <li>{@link SimpleResourceInfo}
  * </ul>
+ *
+ * The return type of the method must be either be of type {@code void}, {@code Optional<Response>}, {@code Uni<Void>} or
+ * {@code Uni<Response>}.
+ * <ul>
+ * <li>{@code void} should be used when filtering does not need to perform any blocking operations and the filter cannot abort
+ * processing.
+ * <li>{@code Optional<Response>} should be used when filtering does not need to perform any blocking operations but the filter
```

Comment:
Why not plain `Response` as a return type?

Reply:
Good question. I thought of making it `Optional` because a filter doesn't have to return it, so I thought I'd make it explicit.

But we can of course support it, it's no big deal.

Reply:
I'd support both, if returning a `Response` and it's `null` then it's a no-abort case.

Reply:
Will do

Reply:
Fixed